core/chains/evm/txmgr/evm_tx_store.go

```diff
@@ -1639,6 +1712,20 @@ func (o *evmTxStore) CountUnconfirmedTransactions(ctx context.Context, fromAddre
 	return o.countTransactionsWithState(ctx, fromAddress, txmgr.TxUnconfirmed, chainID)
 }
 
+// CountAllUnconfirmedTransactions returns the number of unconfirmed transactions for all fromAddresses
+func (o *evmTxStore) CountAllUnconfirmedTransactions(ctx context.Context, chainID *big.Int) (count uint32, err error) {
+	var cancel context.CancelFunc
+	ctx, cancel = o.mergeContexts(ctx)
+	defer cancel()
+	qq := o.q.WithOpts(pg.WithParentCtx(ctx))
+	err = qq.Get(&count, `SELECT count(*) FROM evm.txes WHERE state = $1 AND evm_chain_id = $2`,
+		txmgr.TxUnconfirmed, chainID.String())
+	if err != nil {
+		return 0, fmt.Errorf("failed to CountAllUnconfirmedTransactions: %w", err)
+	}
+	return count, nil
+}
```

Comment:
It may be worth making this method more generic and allowing it to query tx count for any state. You could rename the method to `CountTransactionsByState` and take in a tx state parameter that you get the count for. In case in the future we need to track `fatal_error` tx counts, we could just use the same method